app/src/test/java/de/triology/cas/services/RegistryLocalTest.java

```diff
@@ -502,6 +506,77 @@ public void addDoguChangeListener() throws IOException, InterruptedException {
         assertThat(dogus, hasSize(1));
     }
 
+    @Test
+    public void ChangeListenerWithReInitialization() throws IOException, InterruptedException {
+        var watchKey = mock(WatchKey.class);
+        when(watchKey.isValid())
+                .thenReturn(false)
+                .thenReturn(true)
+                .thenReturn(true);
+        List<WatchEvent<?>> watchEvents = new ArrayList<>();
+        watchEvents.add(new EventMock("local.yaml"));
+        when(watchKey.pollEvents())
+                .thenReturn(watchEvents);
+        when(watchKey.reset())
+                .thenReturn(false)
+                .thenReturn(true);
+        Class<? extends WatchService> wsClass;
+        try (var ws = FileSystems.getDefault().newWatchService()) {
+            wsClass = ws.getClass();
+        }
+        var watchService = mock(wsClass);
+        when(watchService.take())
+                .thenReturn(watchKey)
+                .thenReturn(watchKey)
+                .thenReturn(watchKey)
+                .thenThrow(InterruptedException.class); // finish on fourth invocation
+        var fileSystem = mock(FileSystem.class);
+        when(fileSystem.newWatchService())
+                .thenReturn(watchService)
+                .thenReturn(watchService)
+                .thenReturn(watchService);
+
+        var registry = spy(RegistryLocal.class);
+        registry.fileSystem = fileSystem;
+        var initialServiceAccounts = new RegistryLocal.ServiceAccounts();
+        var changedServiceAccounts = new RegistryLocal.ServiceAccounts();
+        changedServiceAccounts.setCas(Map.of("usermgt", new RegistryLocal.ServiceAccountCas()));
+        var unchangedServiceAccounts2 = new RegistryLocal.ServiceAccounts();
+        unchangedServiceAccounts2.setCas(Map.of("usermgt", new RegistryLocal.ServiceAccountCas()));
+        doReturn(initialServiceAccounts, // get initial state for comparison
+                initialServiceAccounts, // no change
+                changedServiceAccounts, // detect change
+                unchangedServiceAccounts2) // no change to previous state
+                .when(registry).readServiceAccounts();
+
+        ArrayList<String> dogus = new ArrayList<>();
+
+        registry.addDoguChangeListener(() -> {
+            synchronized (dogus) {
+                dogus.add("dogu " + dogus.size());
+                dogus.notify();
+            }
+        });
+
+        synchronized (dogus) {
+            dogus.wait();
+        }
+
+        assertThat(dogus, hasSize(1));
```

Comment:
I don't get this assertion. What is it checking? That the Listener is not changing the amount of dogus?

Reply:
Yes. Just a change after a reinitialisation.